### PR TITLE
Adjust LDS calculation for NGG/NGG GS

### PR DESCRIPTION
**Pull request by amdrexu**
Update the calculation based on internal experiment results.

Change-Id: I7bee987687862c6b4209e24e725eedcffa17d7b7

### CODE REVIEW

File: lgc/patch/PatchResourceCollect.cpp [hunk 1]
```diff
@@ -672,12 +672,18 @@ bool PatchResourceCollect::checkGsOnChipValidity() {
           esVertsPerSubgroup = nggControl->vertsPerSubgroup;
           gsPrimsPerSubgroup = nggControl->primsPerSubgroup;
           break;
-        default:
         case NggSubgroupSizing::Auto:
-          esVertsPerSubgroup = 126;
-          gsPrimsPerSubgroup = 128;
+          if (m_pipelineState->getTargetInfo().getGfxIpVersion() == GfxIpVersion{10, 1}) {
+            esVertsPerSubgroup = 126;
+            gsPrimsPerSubgroup = 128;
```
Comment (linqun) [inline lgc/patch/PatchResourceCollect.cpp:678]: can yo use NggMaxThreadsPerSubgroup / 2 too?

**Review by linqun — APPROVED** [review]
looks good to me for others.

File: lgc/patch/PatchResourceCollect.cpp [hunk 2]
```diff
@@ -753,6 +769,9 @@ bool PatchResourceCollect::checkGsOnChipValidity() {
       // Make sure that we have at least one primitive
       assert(gsPrimsPerSubgroup >= 1);
 
+      // HW has restriction on NGG + Tessellation where gsPrimsPerSubgroup must be >= 3.
+      assert((hasTs == false) || (gsPrimsPerSubgroup >= 3));
```
Comment (kuhar) [inline lgc/patch/PatchResourceCollect.cpp:773]: nit: `assert(!hasTs || gsPrimsPerSubgroup >= 3);`

File: lgc/state/TargetInfo.cpp [hunk 3]
```diff
@@ -69,7 +69,7 @@ static void setGfx6BaseInfo(TargetInfo *targetInfo) {
   targetInfo->getGpuProperty().tessFactorBufferSizePerSe = 4096;
 
   // TODO: Accept gsOnChipDefaultLdsSizePerSubgroup from panel option
-  targetInfo->getGpuProperty().gsOnChipDefaultLdsSizePerSubgroup = 8192; // GFX6-8 value
+  targetInfo->getGpuProperty().gsOnChipDefaultLdsSizePerSubgroup = 8192;
```
Comment (kuhar) [inline lgc/state/TargetInfo.cpp:72]: Why was this comment removed? I guess it applied to all supported GFX targets?

File: lgc/patch/PatchResourceCollect.cpp (hunk 2 above)
Comment (amdrexu) [reply lgc/patch/PatchResourceCollect.cpp:773]: Done

File: lgc/state/TargetInfo.cpp (hunk 3 above)
Comment (amdrexu) [reply lgc/state/TargetInfo.cpp:72]: Yes, when this comment was written, we didn't support GFX9+ on LLPC. Thus, it is outdated now.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6cebd70f27da80ba8460d3fc573250fb9150cb13
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 8abed125acda77ccf700ddd0f5bf0114d25ac135</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev 2f28502c5e6e4f059a014ab30001a191328f7a5d</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 0472e039337a255a85b0f6f4ac927aa19261c2f0</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/220156)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 26609/55154 (48.2%)</li>
<li>Failed: 0/55154 (0.0%)</li>
<li>Skipped: 28545/55154 (51.8%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26609/55154 (48.2%)</li>
<li>Failed: 0/55154 (0.0%)</li>
<li>Skipped: 28545/55154 (51.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 25198/55231 (45.6%)</li>
<li>Failed: 0/55231 (0.0%)</li>
<li>Skipped: 30033/55231 (54.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 26225/54617 (48.0%)</li>
<li>Failed: 0/54617 (0.0%)</li>
<li>Skipped: 28392/54617 (52.0%)</li>
</ul></details>


</ul></details>

**Comment by JaxLinAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit ed5d4f25a75ac41cc02e3fe6cb3e43e36512bbba
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 8abed125acda77ccf700ddd0f5bf0114d25ac135</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev 2f28502c5e6e4f059a014ab30001a191328f7a5d</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 4ad346045fdd6ef7593ce198b78ed563b74b68c5</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/222444)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 26620/55706 (47.8%)</li>
<li>Failed: 0/55706 (0.0%)</li>
<li>Skipped: 29086/55706 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26620/55706 (47.8%)</li>
<li>Failed: 0/55706 (0.0%)</li>
<li>Skipped: 29086/55706 (52.2%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 25209/55783 (45.2%)</li>
<li>Failed: 0/55783 (0.0%)</li>
<li>Skipped: 30574/55783 (54.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 26299/55249 (47.6%)</li>
<li>Failed: 0/55249 (0.0%)</li>
<li>Skipped: 28950/55249 (52.4%)</li>
</ul></details>


</ul></details>

**Pull request merged**